lib/chef/resource/ifconfig.rb

```diff
@@ -35,23 +35,63 @@ class Ifconfig < Chef::Resource
       default_action :add
       allowed_actions :add, :delete, :enable, :disable
 
-      property :target, String, name_property: true
-      property :hwaddr, String
-      property :mask, String
-      property :family, String, default: "inet"
-      property :inet_addr, String
-      property :bcast, String
-      property :mtu, String
-      property :metric, String
-      property :device, String, identity: true
-      property :onboot, String
-      property :network, String
-      property :bootproto, String
-      property :onparent, String
-      property :ethtool_opts, String
-      property :bonding_opts, String
-      property :master, String
-      property :slave, String
+      property :target, String,
+               name_property: true,
+               description: "The IP address that is to be assigned to the network interface. If not specified we'll use the resource's name."
+
+      property :hwaddr, String,
+               description: "The hardware address for the network interface."
+
+      property :mask, String,
+               description: "The decimal representation of the network mask. For example: 255.255.255.0."
+
+      property :family, String,
+               default: "inet", introduced: "14.0",
+               description: "Networking family option for Debian-based systems; for example: inet or inet6."
```

Comment:
Can I be super pedantic that you used two different styles of `for example` here and the one above? O:-)